File: python/checkDivisibility.py
# Time -> 0(N) Space -> 0(1)
def checkDivisibility(n: int) -> bool:
    val = str(n)

    prod, sum = 1, 0
    for s in val:
        num = int(s)

        prod *= num
        sum += num

    return n % (prod + sum) == 0


print(checkDivisibility(99))
print(checkDivisibility(23))

File: go/CheckDivisibility/CheckDivisibility.go
package main

import (
	"fmt"
)

// Time -> 0(N) Space -> 0(1)
func checkDivisibility(n int) bool {
	val := fmt.Sprint(n)
	prod, sum := 1, 0

	for idx := range val {
		num := int(val[idx] - '0')
		prod *= num
		sum += num
	}

	return n%(prod+sum) == 0
}

func main() {
	fmt.Println(checkDivisibility(99))
	fmt.Println(checkDivisibility(23))
}
